app/utils/HouseConsignmentAnswersHelper.scala

```diff
@@ -0,0 +1,241 @@
+      y <- OptionT.liftF(referenceDataService.getCountryNameByCode(x))
+    } yield transportRegisteredCountry(y)).value
+
+  def buildMeansOfTransportRows(idRow: Option[SummaryListRow], nationalityRow: Option[SummaryListRow]): Seq[SummaryListRow] =
+    (idRow, nationalityRow) match {
+      case (Some(transportMeansIDRow), Some(transportRegisteredCountryRow)) => Seq(transportMeansIDRow, transportRegisteredCountryRow)
+      case (Some(transportMeansIDRow), None)                                => Seq(transportMeansIDRow)
+      case (None, Some(transportRegisteredCountryRow))                      => Seq(transportRegisteredCountryRow)
+      case (_, _)                                                           => Seq.empty
+    }
+
+  def buildTransportSections: Future[Seq[Section]] =
+    userAnswers
+      .get(DepartureTransportMeansListSection(houseConsignmentIndex))
+      .traverse {
+        _.zipWithIndex.traverse {
+          y =>
+            val nationalityRow: Future[Option[SummaryListRow]] = buildVehicleNationalityRow(y._2)
+            val meansIdRow: Option[SummaryListRow]             = transportMeansID(y._2)
+
+            nationalityRow.map {
+              nationalityRow =>
+                Section(
+                  messages("unloadingFindings.subsections.transportMeans", y._2.display),
+                  buildMeansOfTransportRows(meansIdRow, nationalityRow)
+                )
+            }
+        }
+      }
+      .map(_.getOrElse(Seq.empty))
+
+  def transportMeansID(transportMeansIndex: Index): Option[SummaryListRow] = getAnswerAndBuildRowWithDynamicPrefix[String](
+    answerPath = DepartureTransportMeansIdentificationNumberPage(houseConsignmentIndex, transportMeansIndex),
+    titlePath = DepartureTransportMeansIdentificationTypePage(houseConsignmentIndex, transportMeansIndex),
+    formatAnswer = formatAsText,
+    dynamicPrefix = formatIdentificationTypeAsText,
+    id = None,
+    call = None
+  )
+
+  def transportRegisteredCountry(answer: String): SummaryListRow = buildSimpleRow(
+    answer = Text(answer),
+    label = messages("unloadingFindings.rowHeadings.vehicleNationality"),
+    prefix = "unloadingFindings.rowHeadings.vehicleNationality",
+    id = None,
+    call = None,
+    args = Seq.empty
+  )
+
+  def houseConsignmentSection: Seq[Section] = {
+
+    val grossAndNetWeightRows: Option[Seq[SummaryListRow]] = houseConsignmentTotalWeightRows
+    val consignorNameRow: Option[SummaryListRow]           = consignorName
+    val consignorIdentificationRow: Option[SummaryListRow] = consignorIdentification
+
+    val rows = buildHouseConsignmentRows(grossAndNetWeightRows, consignorNameRow, consignorIdentificationRow)
+
+    Seq(
+      Section(
+        sectionTitle = messages("unloadingFindings.subsections.houseConsignment", houseConsignmentIndex.display),
+        rows
+      )
+    )
+  }
+
+  private def buildHouseConsignmentRows(
```

Comment:
This method seems quite complicated. Could we maybe do a .getOrElse(Seq.empty) and concatenate them together rather than making a large case statement?